src/graphql/__tests__/entries.test.ts

```diff
@@ -87,6 +90,311 @@ describe('Update GraphQL Entry', () => {
     });
   });
 
+  describe('single reference fields', () => {
+    it('calls sendMessageToEditor when entry being added is not in the entityReferenceMap', () => {
+      const data = {
+        __typename: 'Page',
+        sys: {
+          id: entry.sys.id,
+        },
+        reference: null,
+      };
+      const testAddingEntryId = '18kDTlnJNnDIJf6PsXE5Mr';
+      const sendMessageToEditor = vi.spyOn(Utils, 'sendMessageToEditor').mockReturnThis();
+      const update = {
+        sys: {
+          id: entry.sys.id,
+        },
+        fields: {
+          reference: {
+            'en-US': {
+              sys: {
+                type: 'Link',
+                linkType: 'Entry',
+                id: testAddingEntryId,
+              },
+            },
+          },
+        },
+      } as unknown as EntryProps<KeyValueMap>;
+
+      // value has not changed, just sends message back to editor
+      expect(updateFn({ data, update })).toEqual({
+        __typename: 'Page',
+        sys: {
+          id: entry.sys.id,
+        },
+        reference: null,
+      });
+      expect(sendMessageToEditor).toHaveBeenCalled();
+    });
+
+    it('generates a __typename when entry being added is in the entityReferenceMap then adds this to the modified return value', () => {
+      const data = {
+        __typename: 'Page',
+        sys: {
+          id: entry.sys.id,
+        },
+        reference: null,
+      };
+      const testAddingEntryId = '18kDTlnJNnDIJf6PsXE5Mr';
+      const testContentTypeId = 'testContentType';
+      const update = {
+        sys: {
+          id: entry.sys.id,
+        },
+        fields: {
+          reference: {
+            'en-US': {
+              sys: {
+                type: 'Link',
+                linkType: 'Entry',
+                id: testAddingEntryId,
+              },
+            },
+          },
+        },
+      } as unknown as EntryProps<KeyValueMap>;
+
+      const entityReferenceMap = new Map<string, EntryProps | AssetProps>();
+      entityReferenceMap.set(testAddingEntryId, {
+        sys: {
+          contentType: {
+            sys: {
+              id: testContentTypeId,
+              type: 'TestContentType',
+              linkType: 'Entry',
+            },
+          },
+        },
+      } as EntryProps<KeyValueMap>);
+
+      // value has not changed, just sends message back to editor
+      expect(updateFn({ data, update, entityReferenceMap })).toEqual({
+        __typename: 'Page',
+        sys: {
+          id: entry.sys.id,
+        },
+        reference: {
+          // content type has been adjusted to have capital letter at the start
+          __typename: 'TestContentType',
```

Comment:
it should be the same as in line 165 right?
is the testdata in 164 wrong or the comment or did I get it wrong?  🤯